syntax = "proto3";

import "google/protobuf/any.proto";

message TraceSession {
    repeated TraceEvent events = 1;
    repeated StackFrame stack_frames = 2;
}

message TraceEvent {
    string event_name = 1;
    ExecutionCursor cursor = 2;
    int32 stack_id = 3;
    Variables input_variables = 4;
    Variables locals = 5;
    Variables return_variables = 6;

}
message Variables {
    repeated Variable variables = 1;
}

message Variable {
    string name = 1;
    string value = 2;
}
message ExecutionCursor {
  string file = 1;
  int32 line = 2;
}

message StackFrame {
  int32 id = 1;
  string file = 2;
  int32 line = 3;
    int32 parent_id = 4;
}